### PR TITLE
Fix bug with changing `f._current_pos` when call `f.readline()`

### PR DESCRIPTION
**Pull request by inksink**
Use `_read_from_buffer` to set self._current_pos

### CODE REVIEW

File: smart_open/s3.py [hunk 1]
```diff
@@ -232,12 +232,10 @@ def readline(self, limit=-1):
             #
             if self._line_terminator in self._buffer:
                 next_newline = self._buffer.index(self._line_terminator)
```
Comment (piskvorky) [inline smart_open/s3.py]: Unrelated to this PR, but this seems inefficient: why search for `_line_terminator` twice? Here and in the line above? This is a linear, potentially costly operation, so we should only do it once.

Comment (mpenkov) [reply smart_open/s3.py]: You're right about it being costly, but when I wrote this code, I measured the alternative (try-catch), and that was slower (see the comment above this function). This behavior depends on the data you're processing, and what proportion of reads end up having the line terminator inside the buffer.

**Comment by inksink** [comment]
OK. I will add tests for it.

**Comment by inksink** [comment]
I have changed `test_readline()` to test `tell()` after `readline()`

**Pull request merged**